src/errors.rs

```diff
@@ -1,61 +1,110 @@
 // Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
-use thiserror::Error;
+
+use crate::StdError;
+use alloc::boxed::Box;
+use alloc::string::String;
+use core::fmt;
 
 /// The base error type for raft
-#[derive(Debug, Error)]
+#[derive(Debug)]
```

Comment:
Why was Error removed? Is it not no_std compatible at all? I thought the issue was just with the IO errors.

Perhaps consider one of those instead

- https://crates.io/crates/snafu
- https://crates.io/crates/displaydoc